components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/api/RecoveryApiV2.java

```diff
@@ -0,0 +1,196 @@
+    }
+
+    public RecoveryApiV2(ApiClient apiClient) {
+
+        this.apiClient = apiClient;
+    }
+
+    public ApiClient getApiClient() {
+
+        return apiClient;
+    }
+
+    public void setApiClient(ApiClient apiClient) {
+
+        this.apiClient = apiClient;
+    }
+
+    /**
+     * Initiate recovering the forgotten password.
+     *
+     * @param recoveryInitRequest Password recovery initiating request. (required)
+     * @param tenantDomain        Tenant Domain which user belongs. Default "carbon.super" (optional)
+     * @param headers             If reCaptcha respond is found, embedded in request header. (optional)
+     * @return Account recovery options response object.
+     * @throws ApiException If fails to make API call.
+     */
+    public List<AccountRecoveryType> initiatePasswordRecovery(RecoveryInitRequest recoveryInitRequest,
+                                                              String tenantDomain, Map<String, String> headers)
+            throws ApiException {
+
+        String localVarPath = "/password/init".replaceAll("\\{format\\}", "json");
+        return initiateRecovery(recoveryInitRequest, tenantDomain, headers, localVarPath);
+    }
+
+    /**
+     * Recover password via selected recovery option.
+     *
+     * @param recoveryRequest   Recovery request. (required)
+     * @param tenantDomain      Tenant Domain which user belongs. Default "carbon.super" (optional)
+     * @param headers           Any additional headers to be embedded. (optional)
+     * @return Recovery response.
+     * @throws ApiException If fails to make API call.
+     */
+    public RecoveryResponse recoverPassword(RecoveryRequest recoveryRequest, String tenantDomain,
+                                            Map<String, String> headers) throws ApiException {
+
+        String localVarPath = "/password/recover".replaceAll("\\{format\\}", "json");
+        return recover(recoveryRequest, tenantDomain, headers, localVarPath);
+    }
+
+    /**
+     * This API can be used to initiate recovering forgotten password/username.
+     *
+     * @param recoveryInitRequest Recovery initiating request. (required)
+     * @param tenantDomain        Tenant Domain which user belongs. Default "carbon.super" (optional)
+     * @param headers             If reCaptcha respond is found, embedded in request header. (optional)
+     * @param localVarPath        Endpoint path.
+     * @return Account recovery options response object.
+     * @throws ApiException If fails to make API call.
+     */
+    private List<AccountRecoveryType> initiateRecovery(RecoveryInitRequest recoveryInitRequest, String tenantDomain,
+                                                       Map<String, String> headers, String localVarPath)
+            throws ApiException {
+
+        Object localVarPostBody = recoveryInitRequest;
+        // Verify the required parameter 'recoveryInitRequest' is set.
+        if (recoveryInitRequest == null) {
```

Comment:
What are the scenarios that the `recoveryInitRequest` becomes null? Dont we need to check the details inside that object?

Reply:
Also can you please explain the purpose of `recoveryInitRequest`?

Reply:
> What are the scenarios that the `recoveryInitRequest` becomes null? Dont we need to check the details inside that object?

`recoveryInitRequest` can be null when the request body is not passed. This contains the claims attribute and properties attribute (optional). `claims` attribute has been made mandatory by the API itself (https://github.com/wso2/identity-api-user/blob/master/components/org.wso2.carbon.identity.api.user.recovery/org.wso2.carbon.identity.rest.api.user.recovery.v2/src/gen/java/org/wso2/carbon/identity/rest/api/user/recovery/v2/model/InitRequest.java#L59). 

> Also can you please explain the purpose of `recoveryInitRequest`?

`recoveryInitRequest` consists of the user claims which are used to identify the user and return the available notification channels of the particular user. 

Sample:
```
{
    "claims": [
        {
            "uri": "http://wso2.org/claims/username",
            "value": "rash1"
        },
        {
            "uri": "http://wso2.org/claims/emailaddress",
            "value": "tempabc123@mailinator.com"
        }
    ],
    "properties": [
        {
            "key": "key",
            "value": "value"
        }
    ]
}
```